File: Data Structures & Algorithms/combination-sum-iv/submission-3.cpp
class Solution {
public:
    int f(int target, vector<int>& nums, vector<int>& dp) {
    if(target == 0) return 1;

    if(dp[target] != -1)
        return dp[target];

    int ways = 0;

    for(int x : nums) {
        if(x <= target)
            ways += f(target - x, nums, dp);
    }

    return dp[target] = ways;
}

int combinationSum4(vector<int>& nums, int target) {
    vector<int> dp(target + 1, -1);
    return f(target, nums, dp);
}
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res=0
        st=set(nums)
        longest=0
              
        for num in nums:
            if num-1 not in st:
                len=1
                while(num+len in st):
                    len+=1
                longest=max(longest,len)

        
        return longest
                 